### DISCUSSION TITLE
Sending cached envelopes created by sentry_handle_exception

### DISCUSSION THREAD

**Discussion by hasore** [getsentry/sentry-native#817] [General]
How to drain the cached envelopes created by sentry_handle_exception?
Needs to wait for application restart or there's a better behaviour to send the cached envelopes created?

**Comment by Swatinem** [comment]
Yes, a restart of the application (or just the SDK via `sentry_init`) would be needed, as `sentry_init` will iterate over all the cached envelopes on disk and submit them for sending.

I’m wondering how you ended up in this situation. Envelopes should only be cached on disk from within the signal handler, which operates under the assumption that the application is indeed terminating after handling a hard crash.

**Comment by hasore** [comment]
A custom handler function for exceptions from selected functions.

```c++
LONG __stdcall ExceptCallBack ( EXCEPTION_POINTERS * pExPtrs )
{
#ifdef USE_SENTRY
	if ( !g_pkSentryManager || !g_pkSentryManager->IsRunning() )
		return EXCEPTION_EXECUTE_HANDLER;

	sentry_ucontext_s ex{};
	ex.exception_ptrs = *pExPtrs;
	sentry_handle_exception(&ex);

	return EXCEPTION_EXECUTE_HANDLER;
#else
...
#endif
```
}

```c++
void fn( ... )
{
	__try
	{
...
	}
	__except (ExceptCallBack (GetExceptionInformation()))
	{
...
	}
}
```

But i have already changed the function to flush when a certain number of envelopes are created using sentry_close then sentry_init.
